### PR TITLE
[Fix][Connector-V2] Fix Kudu Source redundant assignment

### PR DESCRIPTION
**Pull request by zhaomin1423**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by Hisoka-X** [comment]
```java
Serializable[][] parameterValues = getParameterValues(partitionParameter.minValue, partitionParameter.maxValue, parallelism);
```
Can you help move this logic into `KuduSourceSplitEnumerator.open()`? Why need execute every time when registerReader() be called?

**Comment by zhaomin1423** [comment]
> ```java
> Serializable[][] parameterValues = getParameterValues(partitionParameter.minValue, partitionParameter.maxValue, parallelism);
> ```
> 
> Can you help move this logic into `KuduSourceSplitEnumerator.open()`? Why need execute every time when registerReader() be called?

Good find, done.

**Comment by Hisoka-X** [comment]
> > ```java
> > Serializable[][] parameterValues = getParameterValues(partitionParameter.minValue, partitionParameter.maxValue, parallelism);
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Can you help move this logic into `KuduSourceSplitEnumerator.open()`? Why need execute every time when registerReader() be called?
> 
> Good find, done.

Great! Can you also help to add e2e? Refenerce: https://github.com/apache/incubator-seatunnel/tree/dev/seatunnel-e2e. 
Thanks!

File: seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/source/KuduSourceSplitEnumerator.java [hunk 1]
```diff
@@ -39,11 +40,19 @@ public class KuduSourceSplitEnumerator implements SourceSplitEnumerator<KuduSour
     public KuduSourceSplitEnumerator(SourceSplitEnumerator.Context<KuduSourceSplit> enumeratorContext, PartitionParameter partitionParameter) {
         this.enumeratorContext = enumeratorContext;
         this.partitionParameter = partitionParameter;
+        this.parallelism = enumeratorContext.currentParallelism();
```
Comment (ashulin) [inline seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/source/KuduSourceSplitEnumerator.java:43]: `EnumeratorContext.currentParallelism()` is Runtime Method, Use it in `SplitEnumerator#run()`

File: seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/source/KuduSourceSplitEnumerator.java [hunk 2]
```diff
@@ -39,11 +40,19 @@ public class KuduSourceSplitEnumerator implements SourceSplitEnumerator<KuduSour
...
     public void open() {
-
+        if (null != partitionParameter) {
+            Serializable[][] parameterValues = getParameterValues(partitionParameter.minValue, partitionParameter.maxValue, parallelism);
+            for (int i = 0; i < parameterValues.length; i++) {
+                allSplit.add(new KuduSourceSplit(parameterValues[i], i));
+            }
+        } else {
+            allSplit.add(new KuduSourceSplit(null, 0));
+        }
```
Comment (ashulin) [inline seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/source/KuduSourceSplitEnumerator.java:55]: Move it into #run()

File: seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/source/KuduSourceSplitEnumerator.java (hunk 1 above)
Comment (ashulin) [reply seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/source/KuduSourceSplitEnumerator.java:43]: Please do not save the value returned by `EnumeratorContext.currentParallelism()`

Comment (zhaomin1423) [reply seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/source/KuduSourceSplitEnumerator.java:43]: thanks, modify it later.

**Pull request closed without merging**